src/main/java/net/snowflake/client/core/bind/BindUploader.java

```diff
@@ -183,25 +194,150 @@ public static synchronized BindUploader newInstance(SFSession session, String st
     }
   }
 
+  /**
+   * Create a new BindUploader which will upload to the given stage path Ensure temporary directory
+   * for file writing exists
+   *
+   * @param session the session to use for uploading binds
+   * @param stageDir the stage path to upload to
+   * @return BindUploader instance
+   */
+  public static synchronized BindUploader newInstanceWithoutDirectory(
+      SFSession session, String stageDir) {
+    return new BindUploader(session, stageDir, null);
+  }
+
   /**
    * Upload the bindValues to stage
    *
    * @param bindValues the bind map to upload
    * @throws BindException if the bind map could not be serialized or upload fails
    */
-  public void upload(Map<String, ParameterBindingDTO> bindValues) throws BindException {
+  @Deprecated
+  public void uploadDeprecated(Map<String, ParameterBindingDTO> bindValues) throws BindException {
     if (!closed) {
       serializeBinds(bindValues);
       putBinds();
     }
   }
 
+  /**
+   * Upload bind parameters via streaming. This replaces previous function upload function where
+   * binds were written to a file which was then uploaded with a PUT statement.
+   *
+   * @param bindValues the bind map to upload
+   * @throws BindException
+   * @throws SQLException
+   */
+  public void upload(Map<String, ParameterBindingDTO> bindValues)
+      throws BindException, SQLException {
+    if (!closed) {
+      List<ColumnTypeDataPair> columns = getColumnValues(bindValues);
+      List<byte[]> csvRows = buildRowsAsBytes(columns);
```

Comment:
why call it `csvRows`? I didn't see comma separated vectors used here.

Reply:
They are csv rows converted to bytes. It happens within the buildRowsAsBytes() function.  I changed it to bindingRows for clarity.